sidecar/src/log.rs

```diff
@@ -44,6 +44,7 @@ where
     live_counter: Mutex<HashMap<K, i32>>,
     pending_removal: Mutex<PriorityQueue<K, Instant>>,
     pub expire_after: Duration,
+    logs_created: Mutex<HashMap<Level, u32>>,
```

Comment:
Wrong place, the datastructure is e.g. used for both filter and logs.
Instead of stuffing it into TemporarilyRetainedMap, please try to keep the logic contained to MultiEnvFilter.

e.g. replace `pub type MultiEnvFilter = TemporarilyRetainedMap<String, EnvFilter>;` by `struct MultiEnvFilter { pub map: TemporarilyRetainedMap<String, EnvFilter>;  logs_created: Mutex<HashMap<Level, u32>>, }`.